R/helper_exec.R

```diff
@@ -23,6 +23,9 @@ future_map = function(n, FUN, ..., MoreArgs = list()) {
   if (getOption("mlr3.debug", FALSE)) {
     lg$info("Running experiments sequentially in debug mode with %i iterations", n)
     mapply(FUN, ..., MoreArgs = MoreArgs, SIMPLIFY = FALSE, USE.NAMES = FALSE)
+  } else if (requireNamespace("mirai", quietly = TRUE) && mirai::status()$connections) {
+    lg$debug("Running resample() via mirai with %i iterations", n)
+    mirai::collect_mirai(mirai::mirai_map(data.table(...), FUN, .args = c(MoreArgs, list(is_sequential = FALSE))))
```

Comment:
why does it call `data.table(...)`? And does it have any chance of getting into conflict with any of the special args of `data.table()` (e.g. `key`, `keep.rownames`)

Reply:
As discussed today: to avoid larger objects to be serialized for every row, use multiple calls to `mirai_map()` + `.args`